apm-sniffer/optional-plugins/undertow-worker-thread-pool-plugin/pom.xml

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one or more
+  ~ contributor license agreements.  See the NOTICE file distributed with
+  ~ this work for additional information regarding copyright ownership.
+  ~ The ASF licenses this file to You under the Apache License, Version 2.0
+  ~ (the "License"); you may not use this file except in compliance with
+  ~ the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>optional-plugins</artifactId>
```

Comment:
Questions, 
1. Should we merge this plugin into the existing undertow tracing plugin? 
2. Should we make it works in default? I can see this is an optional plugin, but a thread pool plugin should not impact the performance, right?

Reply:
About (1), if you want this plugin to be disabled separately from the tracing plugin, I am fine to keep the codes in another plugin jar. But (2) should be considered.

Reply:
@xu1009 Please follow this, then we could merge on both sides.

Reply:
thread pool plugin may impact performance, because it has lock when java get the metric, so it is optional plugin, peoeple really need can add it

Reply:
The fetching period is 20s, I think it's find to keep it in default.
This lock happens in every getting and returning threads from pool. It is not a big impact.
Agree?

Reply:
yes, but the fetching period can be modified by user, if it is set 1s, and the service has high qps, it may impact

Reply:
Manual setting equals optional plugin. So, don't worry.

Reply:
so we make the plugin in default, an add config control it ?

Reply:
Moving it to default is enough. Don't need to add a new config.

Reply:
ok